File: apps/kratos-web/src/enums/ExerciseType.ts
const ExerciseType = {
    Activation: 'activation',
    Class: 'class',
    Endurance: 'endurance',
    Strength: 'strength',
    Stretch: 'stretch',
} as const;

type ExerciseType = typeof ExerciseType[keyof typeof ExerciseType];

export function displayExerciseType(type: ExerciseType): string {
    switch (type) {
        case ExerciseType.Activation:
            return 'Activation';
        case ExerciseType.Class:
            return 'Class';
        case ExerciseType.Endurance:
            return 'Endurance';
        case ExerciseType.Strength:
            return 'Strength';
        case ExerciseType.Stretch:
            return 'Stretch';
    }
}

export default ExerciseType;

File: apis/kratos-api/core/src/enums/exercise_type.rs
use database::impl_bindable;
use serde::{Deserialize, Serialize};
use sqlx::Type;
use strum_macros::Display;

#[derive(Clone, Debug, Deserialize, Display, PartialEq, Serialize, Type)]
#[serde(rename_all = "snake_case")]
#[sqlx(rename_all = "snake_case", type_name = "varchar")]
#[strum(serialize_all = "snake_case")]
pub enum ExerciseType {
    Activation,
    Class,
    Endurance,
    Strength,
    Stretch,
}

impl_bindable!(ExerciseType);

impl Default for ExerciseType {
    fn default() -> Self {
        Self::Strength
    }
}
